aspnetcore/mvc/views/partial.md

```diff
@@ -1,112 +1,152 @@
+The Partial Tag Helper requires ASP.NET Core 2.1 or later. It renders asynchronously and uses an HTML-like syntax:
 
-While there are synchronous equivalents of `Html.PartialAsync` (`Html.Partial`) and `Html.RenderPartialAsync` (`Html.RenderPartial`), use of the synchronous equivalents isn't recommended because there are scenarios where they deadlock. The synchronous methods will be unavailable in future versions.
+[!code-cshtml[](partial/sample/PartialViewsSample/Views/Home/About.cshtml?name=snippet_PartialTagHelper)]
 
-> [!NOTE]
-> If your views need to execute code, the recommended pattern is to use a [view component](view-components.md) instead of a partial view.
+See <xref:mvc/views/tag-helpers/builtin-th/partial-tag-helper> for more information.
+
+::: moniker-end
+
+### Asynchronous HTML Helper
+
+When using an HTML Helper, the best practice is to use [PartialAsync](/dotnet/api/microsoft.aspnetcore.mvc.rendering.htmlhelperpartialextensions.partialasync#Microsoft_AspNetCore_Mvc_Rendering_HtmlHelperPartialExtensions_PartialAsync_Microsoft_AspNetCore_Mvc_Rendering_IHtmlHelper_System_String_). It returns an [IHtmlContent](/dotnet/api/microsoft.aspnetcore.html.ihtmlcontent) type wrapped in a `Task`. The method is referenced by prefixing the call with `@`:
+
+[!code-cshtml[](partial/sample/PartialViewsSample/Views/Home/About.cshtml?name=snippet_PartialAsync)]
+
+You can alternatively render a partial view with [RenderPartialAsync](/dotnet/api/microsoft.aspnetcore.mvc.rendering.htmlhelperpartialextensions.renderpartialasync). This method doesn't return a result. It streams the rendered output directly to the response. Because the method doesn't return a result, it must be called within a Razor code block:
+
+[!code-cshtml[](partial/sample/PartialViewsSample/Views/Home/About.cshtml?name=snippet_RenderPartialAsync)]
+
+Since it streams the result directly, `RenderPartialAsync` may perform better in some scenarios. However, it's recommended you use `PartialAsync`.
+
+### Synchronous HTML Helper
 
-### Partial View Discovery
+[Partial](/dotnet/api/microsoft.aspnetcore.mvc.rendering.htmlhelperpartialextensions.partial) and [RenderPartial](/dotnet/api/microsoft.aspnetcore.mvc.rendering.htmlhelperpartialextensions.renderpartial) are the synchronous equivalents of `PartialAsync` and `RenderPartialAsync`, respectively. Use of the synchronous equivalents isn't recommended because there are scenarios in which they deadlock. Future releases won't contain the synchronous methods.
 
-When referencing a partial view, you can refer to its location in several ways:
+> [!IMPORTANT]
+> If your views need to execute code, use a [view component](xref:mvc/views/view-components) instead of a partial view.
+
+### Partial view discovery
+
+When referencing a partial view, you can refer to its location in several ways. For example:
```

Comment:
Feature the partial tag helper
    `<partial name="_CookieConsentPartial" />`

We need to show that and that's it's the preferred approach.